junit5/src/main/java/org/jboss/weld/junit5/auto/ClassScanning.java

```diff
@@ -101,7 +101,7 @@ static void scanForRequiredBeanClasses(List<Class<?>> testClasses, Weld weld, bo
                     .forEach(cls -> addClassesToProcess(classesToProcess, cls));
 
             AnnotationSupport.findAnnotatedMethods(currClass, Inject.class, HierarchyTraversalMode.BOTTOM_UP).stream()
-                    .map(Method::getReturnType)
```

Comment:
Honestly, I haven't understood the purpose of `getReturnType` as it was there before. I also can't find initializer method return types mentioned in the specs (https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#declaring_initializer). Could it possibly have been a bug, here, to evaluate the return type or shall that be kept and if so what for?
It may not hurt if it works now, regardless of how the test is instantiated.

Reply:
I didn't understand it either and did some digging too - I think it's a bug/oversight from when the annotation based approach was added.
In CDI, `@Inject` usage on methods is basically either initializer or annotated constructor neither of which has interesting return type that we should process here.